### PR TITLE
Sonarqube fixes for LimeSurvey 5

### PR DESCRIPTION
**Pull request by c-schmitz**

### CODE REVIEW

File: application/controllers/UploaderController.php [hunk 1]
```diff
@@ -56,23 +56,23 @@ public function run($actionID)
             // If one seems to be a hack: Bad request
             throw new CHttpException(400); // See for debug > 1
         }
-        if ($sFileGetContent) {
-            if (substr($sFileGetContent, 0, 6) == 'futmp_') {
+        if ($sFileGetContentFiltered) {
```
Comment (Shnoulle) [inline application/controllers/UploaderController.php:59]: Happy to know sonarcloud is OK with this :+1:

File: application/libraries/BigData.php [hunk 2]
```diff
@@ -230,9 +230,11 @@ public function render($type = null)
         if (!isset($type)) {
             $type = $this->defaultEcho;
         }
-        if (method_exists($this, "echo_{$type}")) {
-            call_user_func(array($this, "echo_{$type}"));
+        // TODO: No other types supported, ever?
```
Comment (Shnoulle) [inline application/libraries/BigData.php:233]: 🤣

File: application/controllers/admin/ParticipantsAction.php [hunk 3]
```diff
@@ -1463,7 +1463,15 @@ public function changeAttributeEncrypted()
                         $aUpdateData['value'] = LSActiveRecord::encryptSingle($attribute->value);
                     }
                     if (!empty($aUpdateData) && $aUpdateData['value'] !== null) {
-                        $oDB->createCommand()->update('{{participant_attribute}}', $aUpdateData, "attribute_id='" . $attributeId . "' AND participant_id = '" . $attribute->participant_id . "'");
+                        $oDB->createCommand()->update(
+                            '{{participant_attribute}}',
+                            $aUpdateData,
+                            'attribute_id=:attribute_id AND participant_id=:participant_id',
+                            array(
+                                'attribute_id' => $attributeId,
```
Comment (olleharstedt) [inline application/controllers/admin/ParticipantsAction.php:1471]: Not sure if correct. Should be
```php
':attribute_id' => $attributeId,
```
?

**Review by olleharstedt — CHANGES_REQUESTED** [review]

File: application/controllers/admin/ParticipantsAction.php (hunk 3 above)
Comment (olleharstedt) [reply application/controllers/admin/ParticipantsAction.php:1471]: It's about colon or not in the array key `:attribute_id` vs `attribute_id` :)

Comment (Shnoulle) [reply application/controllers/admin/ParticipantsAction.php:1471]: > It's about colon or not in the array key `:attribute_id` vs `attribute_id` :)

Yes : i think you're right ! Totally right !

Seems not tested a lot …

Comment (fosteruk) [reply application/controllers/admin/ParticipantsAction.php]: Well spotted @olleharstedt. Fix was pushed to the sonarcubefixes branch.

File: application/models/Template.php [hunk 4]
```diff
@@ -134,6 +134,8 @@ public function primaryKey()
      */
     public static function templateNameFilter($sTemplateName)
     {
+        $sTemplateName = sanitize_filename($sTemplateName, false, false, false);
```
Comment (Shnoulle) [inline application/models/Template.php:137]: Seems to work !
Great !!!!

**Pull request merged**